### ISSUE TITLE
helm-controller cannot set --kubeconfig flag

### ISSUE THREAD

**Issue by Poor12** [fluxcd/helm-controller#481]
I have a command that helm-controller is deployed in multicluster scene, just like out of cluster. Seems that helm-controller is based on controller-runtime. But why helm-controller cannot set --kubeconfig flag?
![image](https://user-images.githubusercontent.com/26862112/167785995-661acff0-12b4-4002-8ffb-77c2b13bd300.png)

**Comment by aryan9600** [comment]
You can specify the kubeconfig of the remote cluster in your `HelmRelease` definition, please see: https://fluxcd.io/docs/components/helm/helmreleases/#remote-clusters--cluster-api

**Comment by Poor12** [comment]
Thanks. What you said is about installing helm releases in another cluster. But what if the CRD and the controller is not in the same cluster, controller even cannot watch `helmRelease`. Actually, controller-runtime provides the config by default. I wonder why helm-controller has not.

**Comment by stefanprodan** [comment]
The controller-runtime client does support this:

> KUBECONFIG environment variable pointing at a file

**Comment by aryan9600** [comment]
IMO, we could add support for this, since controller-runtime supports the flag as well: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/config/config.go#L37-L41. It uses the go provided `flag` package and not `pflag`, so we'd need to modify how we parse our flags.

**Comment by stefanprodan** [comment]
Using the `flag` package is not an option for us. And this can't be done here in helm-controller but globally for all controllers in `fluxcd/pkg/runtime/client`.

**Comment by Poor12** [comment]
> IMO, we could add support for this, since controller-runtime supports the flag as well: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/config/config.go#L37-L41. It uses the go provided `flag` package and not `pflag`, so we'd need to modify how we parse our flags.

Yeah, you are right. Off course environment variable make sense. But I think inheriting upstream is better.

**Comment by hiddeco** [comment]
Think configuration of this is already possible by simply setting a `KUBECONFIG` environment variable on the pod.

**Comment by kingdonb** [comment]
@Poor12 Did you try the suggestion, to add a `KUBECONFIG` env variable directly to the controller deployment?

Please let us know if this works for you so that we can close the issue 👍

**Comment by Poor12** [comment]
It works. But actually my suggestion is that flags from controller need to be modified.
